RestaurantsWeb/Controllers/RestaurantsController.cs

```diff
@@ -1,33 +1,71 @@
-﻿using Microsoft.AspNetCore.Mvc;
-using RestaurantsWeb.Repositories;
-using System;
+﻿using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
+using FluentValidation;
+using Microsoft.AspNetCore.Mvc;
+using RestaurantsWeb.Entities;
+using RestaurantsWeb.Models;
 
 namespace RestaurantsWeb.Controllers
 {
     [Route("api/[controller]")]
     [ApiController]
     public class RestaurantsController : ControllerBase
     {
-        private readonly IRestaurantRepository restaurantRepository;
+        private readonly IRestaurantRepository repository;
+        private readonly IValidator<AddRestaurantRequest> validator;
+        private readonly IValidator<UpdateRestaurantRequest> updateValidator;
 
-        public RestaurantsController(IRestaurantRepository restaurantRepository)
+        public RestaurantsController(IRestaurantRepository repository, 
+            IValidator<AddRestaurantRequest> validator,
+            IValidator<UpdateRestaurantRequest> updateValidator
+
+            )
+        {
+            this.repository = repository;
+            this.validator = validator;
+            this.updateValidator = updateValidator;
+        }
+
+        public IActionResult GetAllRestaurants()
+        {
+            var restaurants = repository.GetAllRestaurants();
+            return Ok(restaurants);
+        }
+
+        [HttpPost]
+        public IActionResult AddRestaurant([FromBody] AddRestaurantRequest restaurant)
         {
-            this.restaurantRepository = restaurantRepository;
+            validator.ValidateAndThrow(restaurant);
+            var addRestaurant = repository.AddRestaurant(restaurant);
+            return Ok(addRestaurant);
         }
 
-        // GET api/values
         [HttpGet("{name}")]
-        public IActionResult Get([FromRoute] string name)
+        public IActionResult GetRestaurant([FromRoute] string name)
         {
-            var restaurant = restaurantRepository.GetRestaurant(name);
-            if (restaurant == null)
+            var getRestaurant = repository.GetRestaurant(name);
+            if (getRestaurant == null)
             {
-                return NoContent();
+                throw new Exception("Restaurant not found");
             }
-            return Ok(restaurant); 
+            return Ok(getRestaurant);
+        }
+
+        [HttpPut("{id}")]
+        public IActionResult UpdateRestaurant([FromBody] UpdateRestaurantRequest updateRestaurant, [FromRoute] long id)
```

Comment:
Please change it and put from body to be last and from route to be at the beginning